http-client/src/test/groovy/io/micronaut/http/client/retry/HttpClientRetryWithFallbackSpec.groovy

```diff
@@ -49,27 +52,33 @@ class HttpClientRetryWithFallbackSpec extends Specification{
 
     void "test simple blocking retry"() {
         given:
+        CountFilter countFilter = context.getBean(CountFilter)
         CountClient countClient = context.getBean(CountClient)
         CountController controller = context.getBean(CountController)
         controller.countThreshold = 3
 
         when:"A method is annotated retry"
         int result = countClient.getCount()
 
-        then:"It executes until successful"
+        then:"It executes new requests until successful"
         result == 3
+        result == countFilter.requests.size()
 
         when:"The threshold can never be met"
+        countFilter.requests.clear()
         controller.countThreshold = Integer.MAX_VALUE
         controller.count = 0
         countClient.getCount()
 
         then:"The fallback is called"
         countClient.getCount() == 9999
+        countFilter.requests.size() == 12 //TODO - This seems like a bug in the way blocking retry works with fallback, but is the same as the behavior before HttpClientIntroductionAdvice refactoring
```

Comment:
how much do you expect here?

Reply:
I was thinking it should be 6, the same as it is now in the reactive case below this.

My logic is: 1 initial failed attempt, and 5 failing retries, then it falls back to CountClientFallback, which does not actually initiate an HTTP request.

Reply:
@jeremyg484 please, investigate it and fix it in a separate PR.

Reply:
@jeremyg484 @sdelamo It's running it twice

Once on line 71 and then again in the `then` block:

https://github.com/micronaut-projects/micronaut-core/blob/988c60262c92129db4566a5f11d3548b99bd4bed/http-client/src/test/groovy/io/micronaut/http/client/retry/HttpClientRetryWithFallbackSpec.groovy#L71-L75

If you change it to:

```groovy
        result = countClient.getCount()

        then:"The fallback is called"
        result == 9999
        countFilter.requests.size() == 6
```

Then it's 6 as expected

Reply:
🤦‍♂️ @timyates good catch!

Reply:
I love a good mystery 🕵️